starting on hippo001 (#1)
hemilabs/hippos: hippo001.md

# HIPPO001

## Summary

> ```

Reviewer: As the preamble is essentially the summary, do we want a Summary header here? (Also `HIPPO001` or `HIPPO-001`?)
```suggestion
# HIPPO-001

```

Author: Going with the - in the name and dropping summary as I don't think it makes sense.

HIPPO: 1
Title: HIPPO Purpose and Guidelines
Author: John C. Vernaleo  <jcv@hemi.xyz>
Comments-Summary: No comments yet.
Comments-URI: NA
Status: New
Type: Process
Created: 2025-10-14
Superseded-By: NA
```

## What is a HIPPO

HIPPO is a Hemi Improvement ProPOsal.  These are design documents to describe changes to the Hemi Network.

## HIPPO Process

> All major changes and updates to the Hemi Network should have an accompanying HIPPO that describes the change, steps need for infrastructure runners, and any user visable changes.

Reviewer 2: ```suggestion
All major changes and updates to the Hemi Network should have an accompanying HIPPO that describes the change, the steps needed for infrastructure runners, and any user-visible changes.
```

HIPPOs should be submitted BEFORE a change goes live.

> HIPPOs are numbered in the order they are recieved in this repository.  The order in no way indicated the order or status of the actual changes.  The status field of the HIPPO is what determines if it has actually been deployed.

Reviewer 2: ```suggestion
HIPPOs are numbered in the order they are received in this repository.  The order in no way indicated the order or status of the actual changes.  The status field of the HIPPO is what determines if it has actually been deployed.
```

> Outside of status updates and minor changes (typos, cosmetic, etc.), HIPPOs should not be changed once made active.  They should always be superceded by another HIPPO in the case that a significant change is needed.

Reviewer 2: ```suggestion
Outside of status updates and minor changes (typos, cosmetic, etc.), HIPPOs should not be changed once made active.  They should always be superseded by another HIPPO in the case that a significant change is needed.
```

## HIPPO Types

The following types of HIPPOs are prossible:

* Process:  These describe some change in the way updates to the Hemi Network or the HIPPO process itself work.
* Network:  These describe some change to the Hemi Network itself.  This can range from a major change to the daemon that run Hemi to follow a new Ethereum hardfork.
* Misc:  These are proposals that do not cleanly fall into one of the other categories.

## HIPPO Work Flow

> All HIPPOS must be submitted to this [repository](https://github.com/hemilabs/hippos) as pull requests.  They must be in Markdown and should follow the format of this file and MUST include a summary section containing:

Reviewer: > The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).

?

Author: Added.

```
HIPPO:
Title:
Author:
Comments-Summary:
Comments-URI:
Status:
Type:
Created:

> Superseded-By:

Reviewer: Similar to BIP-0001/BIP-0002, should we have a description for each field and indicate which fields are optional? For example (adding `Requires`, `Replaces`):

```suggestion
  HIPPO: <HIPPO number, or "?" before being assigned>
  Title: <HIPPO short title>
  Author: <list of authors' names and email addresses>
* Comments-Summary: <summary>
  Comments-URI: <link to comments>
  Status: <Draft | Active | Proposed | Rejected | Withdrawn | Replaced>
  Type: <type>
  Created: <creation date, in ISO 8601 (yyyy-mm-dd) format>
* Requires: <HIPPO number(s)>
* Replaces: <HIPPO number>
* Superseded-By: <HIPPO number>  
```

As there could be multiple HIPPOs proposed in PRs at the same time, we could also end up with duplicate numbers - so using `?` before assigned may be beneficial.

Reviewer: Also BIPs seem to use Draft -> Proposed -> (Active | Rejected | Withdrawn) -> Replaced.

Is `New` essentially the same as `Draft`, and is it useful to have both?

https://github.com/bitcoin/bips/blob/master/bip-0002.mediawiki#bip-header-preamble

Author: I expect a small enough number that for now I think we can set number at the time of creation and if we get dupe, deal as we review.  If we are ever end up with enough that it is a problem we can define a way to assign numbers.  I really want this lite on process for now.

Author: All addressed.

```

> When initally submitted, the status should be `New`.

Reviewer 2: ```suggestion
When initially submitted, the status should be `New`.
```
